asyncpg/pool.py

```diff
@@ -507,35 +545,64 @@ def acquire(self, *, timeout=None):
             # Already released, do nothing.
             return
 
-        con = connection._detach()
+        self._check_init()
+
+        con = connection._con
         con._on_release()
```

Comment:
`connection._con._on_release()` to make it easier for the reader (o/w I need to track if `con` is used in the code below)